internal/provider/peer_data_source.go

```diff
@@ -0,0 +1,193 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package provider
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	netbird "github.com/netbirdio/netbird/management/client/rest"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var _ datasource.DataSource = &PeerDataSource{}
+
+func NewPeerDataSource() datasource.DataSource {
+	return &PeerDataSource{}
+}
+
+// PeerDataSource defines the data source implementation.
+type PeerDataSource struct {
+	client *netbird.Client
+}
+
+func (d *PeerDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_peer"
+}
+
+func (d *PeerDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
```

Comment:
Might be worth creating a `_list` equivalent for Peers so we can actually list and act on those.

Reply:
Resolving in https://github.com/netbirdio/terraform-provider-netbird/pull/7